dev/com.ibm.ws.messaging.open_jms20deliverydelay_fat/fat/src/com/ibm/ws/messaging/jms20/deliverydelay/fat/DelayFullTest.java

```diff
@@ -0,0 +1,584 @@
+
+        runInServlet("testDeliveryDelayForDifferentDelays");
+
+        String msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : QueueBindingsMessage2");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : QueueBindingsMessage1");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+
+        restartServers(MDB_CONFIG_QUEUE_TCP);
+
+        runInServlet("testDeliveryDelayForDifferentDelays_Tcp");
+
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : QueueTCPMessage2");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : QueueTCPMessage1");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+
+        restartServers();
+    }
+
+    @Test
+    public void testDeliveryDelayForDifferentDelaysTopic_B() throws Exception {
+        restartClient(MDB_CONFIG_TOPIC_BINDINGS);
+
+        runInServlet("testDeliveryDelayForDifferentDelaysTopic");
+
+        String msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : TopicBindingsMessage2");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : TopicBindingsMessage1");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+
+        restartServers(MDB_CONFIG_TOPIC_TCP);
+
+        runInServlet("testDeliveryDelayForDifferentDelaysTopic_Tcp");
+
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : TopicTCPMessage2");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+        msg = clientServer.waitForStringInLogUsingLastOffset("Message received on mdb : TopicTCPMessage1");
+        assertNotNull("Could not find the upload message in the trace.log", msg);
+
+        restartServers();
+    }
+
+    @Test
+    public void testPersistentMessageStore_B() throws Exception {
+        // dz restartServers(DEFAULT_CLIENT);
+
+        runInServlet("testPersistentMessage");
+
+        restartServers();
+
+        boolean testResult = runInServlet("testPersistentMessageReceive");
+        assertTrue("testPersistentMessageStore_B failed", testResult);
+    }
+
+    @Test
+    public void testPersistentMessageStore_Tcp() throws Exception {
+        // dz restartServers(DEFAULT_CLIENT);
+
+        runInServlet("testPersistentMessage_Tcp");
+
+        restartServers();
+
+        boolean testResult = runInServlet("testPersistentMessageReceive_Tcp");
+        assertTrue("testPersistentMessageStore_Tcp failed", testResult);
+    }
+
+    // TODO: SIB runtime is missing a doPriv.  Test fails without this permission 
```

Comment:
I'll remove this in another branch which repairs the lacking doPriv in the messaging runtime.

Reply:
Noted.  I added to the comment that this would be fixed in a subsequent update.